metagraph/src/graph/annotated_dbg.hpp

```diff
@@ -61,7 +71,8 @@ class AnnotatedSequenceGraph {
 
   protected:
     std::shared_ptr<SequenceGraph> graph_;
-    std::unique_ptr<Annotator> annotator_;
+    std::unique_ptr<Annotation> annotation_;
+    const std::shared_ptr<Annotator> annotator_;
```

Comment:
Will it work if you pass lazy annotator to AnnotatedDBG? This points to a design issue.
Maybe it's better to do without AnnotatedDBG in differential assembly? Better pass graph and annotation separately. Then you can pass lazy annotation in any type you want.